src/main/java/com/iri/dao3/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.iri.dao3.dao.impl;
+
+import com.iri.dao3.dao.ManufacturerDao;
+import com.iri.dao3.db.Storage;
+import com.iri.dao3.lib.Dao;
+import com.iri.dao3.model.Manufacturer;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        return Storage.addManufacturer(manufacturer);
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.manufacturers.stream()
+                .filter(manufacturer -> manufacturer.getId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        return Storage.manufacturers.stream()
+                .filter(manufacturer1 -> manufacturer1.getId().equals(manufacturer.getId()))
+                .map(manufacturer1 -> manufacturer1 = manufacturer)
+                .findFirst()
+                .get();
```

Comment:
```suggestion
        IntStream.range(0, Storage.manufacturers.size())
                .filter(i -> Storage.manufacturers.get(i).getId().equals(manufacturer.getId()))
                .forEach(i -> Storage.manufacturers.set(i, manufacturer));
        return manufacturer;
```
the point of that screenshot was that method update should be rewritten

Reply:
done
Can I do merge?